modules/core/src/main/java/org/apache/ignite/internal/processors/continuous/GridContinuousProcessor.java

```diff
@@ -2056,13 +2098,16 @@ else if (log.isDebugEnabled())
         private boolean autoUnsubscribe;
 
         /**
+         * @param nodeId Node id.
          * @param prjPred Projection predicate.
          * @param hnd Continuous routine handler.
          * @param bufSize Buffer size.
          * @param interval Interval.
          * @param autoUnsubscribe Automatic unsubscribe flag.
          */
-        LocalRoutineInfo(@Nullable IgnitePredicate<ClusterNode> prjPred,
+        LocalRoutineInfo(
+            UUID nodeId,
```

Comment:
We can't dot it because `LocalRoutineInfo` is serializable and sent to remote node.
Please, take a look at `GridContinuousProcessor#getDiscoveryData`